### ISSUE TITLE
run_sft.sh出现eval_loss和perplexity出现nan

### ISSUE THREAD

**Issue by zzx16045223** [ymcui/Chinese-LLaMA-Alpaca-2#191]
### 提交前必须检查以下项目

- [X] 请确保使用的是仓库最新代码（git pull），一些问题已被解决和修复。
- [X] 我已阅读[项目文档](https://github.com/ymcui/Chinese-LLaMA-Alpaca-2/wiki)和[FAQ章节](https://github.com/ymcui/Chinese-LLaMA-Alpaca-2/wiki/常见问题)并且已在Issue中对问题进行了搜索，没有找到相似问题和解决方案。
- [X] 第三方插件问题：例如[llama.cpp](https://github.com/ggerganov/llama.cpp)、[LangChain](https://github.com/hwchase17/langchain)、[text-generation-webui](https://github.com/oobabooga/text-generation-webui)等，同时建议到对应的项目中查找解决方案。

### 问题类型

模型训练与精调

### 基础模型

Alpaca-2-13B

### 操作系统

Linux

### 详细描述问题

将fp16设置为了False，并且将bf16设置为了True，还是出现eval_loss为nan的结果
lr=1e-4
lora_rank=64
lora_alpha=128
lora_trainable="q_proj,v_proj,k_proj,o_proj,gate_proj,down_proj,up_proj"
modules_to_save="embed_tokens,lm_head"
lora_dropout=0.05

pretrained_model=/workspace/ypg/Chinese-LLaMA-Alpaca-2/llama2_model/chinese-alpaca-2-13b
chinese_tokenizer_path=/workspace/ypg/Chinese-LLaMA-Alpaca-2/llama2_model/chinese-alpaca-2-13b
dataset_dir=/workspace/ypg/Chinese-LLaMA-Alpaca-2/dataset/cailian_data/test
per_device_train_batch_size=1
per_device_eval_batch_size=1
gradient_accumulation_steps=8
output_dir=/workspace/ypg/Chinese-LLaMA-Alpaca-2/output/20230824_1000_data/test
# peft_model=/workspace/ypg/Chinese-LLaMA-Alpaca-2/llama2_model/chinese-alpaca-2-13b
validation_file=/workspace/ypg/Chinese-LLaMA-Alpaca-2/dataset/cailian_data/test/output.json

num_train_epochs=0.5
max_seq_length=5000

deepspeed_config_file=ds_zero2_no_offload.json

CUDA_VISIBLE_DEVICES=1 torchrun --nnodes 1 --nproc_per_node 1 --master_port 29502 run_clm_sft_with_peft.py \
    --deepspeed ${deepspeed_config_file} \
    --model_name_or_path ${pretrained_model} \
    --tokenizer_name_or_path ${chinese_tokenizer_path} \
    --dataset_dir ${dataset_dir} \
    --validation_split_percentage 0.001 \
    --per_device_train_batch_size ${per_device_train_batch_size} \
    --per_device_eval_batch_size ${per_device_eval_batch_size} \
    --do_train \
    --do_eval \
    --seed $RANDOM \
    --fp16 False \
    --bf16 True \
    --num_train_epochs ${num_train_epochs} \
    --lr_scheduler_type cosine \
    --learning_rate ${lr} \
    --warmup_ratio 0.03 \
    --weight_decay 0 \
    --logging_strategy steps \
    --logging_steps 10 \
    --save_strategy steps \
    --save_total_limit 3 \
    --evaluation_strategy steps \
    --eval_steps 1000 \
    --save_steps 1000 \
    --gradient_accumulation_steps ${gradient_accumulation_steps} \
    --preprocessing_num_workers 8 \
    --max_seq_length ${max_seq_length} \
    --output_dir ${output_dir} \
    --overwrite_output_dir \
    --ddp_timeout 30000 \
    --logging_first_step True \
    --lora_rank ${lora_rank} \
    --lora_alpha ${lora_alpha} \
    --trainable ${lora_trainable} \
    --modules_to_save ${modules_to_save} \
    --lora_dropout ${lora_dropout} \
    --torch_dtype float16 \
    --validation_file ${validation_file} \
    --gradient_checkpointing \
    --ddp_find_unused_parameters False \
    --flash_attn

    # --peft_path ${peft_model} \

### 依赖情况（代码类问题务必提供）

peft                      0.3.0.dev0
sentence-transformers     2.2.2
torch                     2.0.1
torchvision               0.15.2
transformers              4.31.0

### 运行日志或截图

***** eval metrics *****
  epoch                   =       0.67
  eval_loss               =        nan
  eval_runtime            = 0:00:04.67
  eval_samples            =         12
  eval_samples_per_second =      2.564
  eval_steps_per_second   =      2.564
  perplexity              =        nan

**Comment by airaria** [comment]
只有最后是nan，还是整个训练过程中eval_loss都是nan?

**Comment by zzx16045223** [comment]
> 只有最后是nan，还是整个训练过程中eval_loss都是nan?

在整个训练过程中eval-loss都是nan，请问有什么解决办法吗？

**Comment by airaria** [comment]
你可以用tensorboard看一下output_dir下的训练日志，看看各指标training loss是否正常，模型是否正常收敛，以确定是数据问题还是训练问题。

**Comment by Teeemio** [comment]
可能是中间参数的lm_head的modules_to_save是nan

**Comment by tkone2018** [comment]
@zzx16045223 你好，请问你解决了吗，我也遇到这个问题了

**Comment by tkone2018** [comment]
@Teeemio 我也去掉了modules_to_save这个参数

**Comment by 5663015** [comment]
可以参考：https://github.com/ymcui/Chinese-LLaMA-Alpaca-2/issues/424
